src/main/java/seedu/duke/logic/parser/AddCommandParser.java

```diff
@@ -68,69 +66,78 @@ private static Command parseAddLessonCommand(String userResponse) throws ParseEx
             throw new ParseException(Messages.ERROR_INVALID_TIME_FORMAT);
         }
 
-        String meetingUrl = "-";
         switch (params.length) {
         case 4:
-            if (!hasCorrectFlagSequence(userResponse, "-d", "-s", "-e")) {
-                throw new ParseException(Messages.ERROR_INVALID_FLAG_SEQUENCE);
-            }
-            return new AddLessonCommand(title, dayOfTheWeek, startTime, endTime, meetingUrl);
+            return parseAddLessonFourParams(userResponse, title, dayOfTheWeek, startTime, endTime);
         case 5:
-            if (!hasCorrectFlagSequence(userResponse, "-d", "-s", "-e", "-l")) {
-                throw new ParseException(Messages.ERROR_INVALID_FLAG_SEQUENCE);
-            }
-            meetingUrl = params[4].strip();
-            return new AddLessonCommand(title, dayOfTheWeek, startTime, endTime, meetingUrl);
+            return parseAddLessonFiveParams(userResponse, params[4], title, dayOfTheWeek, startTime, endTime);
         default:
             throw new ParseException(Messages.ERROR_INVALID_COMMAND);
         }
     }
 
+    private static Command parseAddLessonFiveParams(String userResponse, String param, String title,
+                                                    String dayOfTheWeek, String startTime, String endTime)
+            throws ParseException {
+        String meetingUrl;
+        if (!hasCorrectFlagSequence(userResponse, "-d", "-s", "-e", "-l")) {
+            throw new ParseException(Messages.ERROR_INVALID_FLAG_SEQUENCE);
+        }
+        meetingUrl = param.strip();
+        return new AddLessonCommand(title, dayOfTheWeek, startTime, endTime, meetingUrl);
+    }
+
+    private static AddLessonCommand parseAddLessonFourParams(String userResponse, String title, String dayOfTheWeek,
+                                                             String startTime, String endTime) throws ParseException {
+        if (!hasCorrectFlagSequence(userResponse, "-d", "-s", "-e")) {
+            throw new ParseException(Messages.ERROR_INVALID_FLAG_SEQUENCE);
+        }
+        return new AddLessonCommand(title, dayOfTheWeek, startTime, endTime, EMPTY_INFORMATION);
+    }
+
     private static Command parseAddTaskCommand(String userResponse) throws ParseException {
-        String[] params = userResponse.split(" -d | -p | -i");
-        if (params.length < 2 || params.length > 4) {
+        String[] params = userResponse.split(TASK_FLAGS);
+        checkParamsLength(params, 2, 4);
+
+        String title = params[0].strip();
+        String dayOfTheWeek = parseDayOfTheWeek(params[1]);
+
+        switch (params.length) {
+        case 2:
+            return new AddTaskCommand(title, dayOfTheWeek, DEFAULT_PRIORITY, EMPTY_INFORMATION);
+        case 3:
+            return parseAddTaskThreeParams(userResponse, params, title, dayOfTheWeek);
+        case 4:
+            return parseAddTaskFourParams(userResponse, params, title, dayOfTheWeek);

```

Comment:
Same issue with the parse lesson implementation. Having numbers in a method name is usually not a good convention.

Reply:
Consider combining the two methods into one, and move them to `ParserUtil`.